crates/llm_gateway/src/stream_context.rs

```diff
@@ -158,6 +194,20 @@ impl HttpContext for StreamContext {
         );
 
         self.request_id = self.get_http_request_header(REQUEST_ID_HEADER);
+
+        //start the timing for the request using get_current_time()
+        match get_current_time() {
+            Ok(current_time) => {
+                self.start_time = Some(current_time);
+                self.ttft_recorded = false;
+                self.ttft_duration = None;
+            }
+            Err(e) => {
+                warn!("Failed to get current time: {:?}", e);
+                self.start_time = None;
+            }
+        }
```

Comment:
It is unlikely that get_current_time would fail - but it failed we should fail the request